setup.py

```diff
@@ -49,6 +49,8 @@
     'boto3>=1.4,<1.5',
     'freezegun>=0.3.8',
     'openpyxl>=2.6.4',
+    'azure-mgmt-cdn',
```

Comment:
Could you version checks here?

Reply:
I would very much like for us to move for optional dependencies where possible.
i.e. if you need the Azure CDN backend you'd do `pip install wagtail[cdn-azure]` or similar?

This is probably a large discussion to have, so not a blocker on this per se

Reply:
Yep, I agree. We probably want to do something similar for cloudflare, elasticsearch, boto3, etc. Shall we tackle all these in one go in a separate PR?

Reply:
@kaedroho  / @zerolab  I have split this out to a new issue, please can either of you update with a bit more context https://github.com/wagtail/wagtail/issues/7753

Reply:
Note: the original issue raised has been resolved